From 7c3aa14789ff6d0c5ed7f790d2686bf102a07624 Mon Sep 17 00:00:00 2001
Message-ID: <7c3aa14789ff6d0c5ed7f790d2686bf102a07624.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:22 +0300
Subject: [PATCH 01/15] gnu: Add go-github-com-atotto-clipboard.

* gnu/packages/golang-xyz.scm (go-github-com-atotto-clipboard): New variable.

Change-Id: I4a871983f564034045f4b2ba77f2453cfce5cb2b
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 37 ++++++++++++++++++++++++++++++++++++-
 1 file changed, 36 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 0ef4c46d9f..63ac100349 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -85,7 +85,9 @@ (define-module (gnu packages golang-xyz)
   #:use-module (gnu packages golang-maths)
   #:use-module (gnu packages golang-web)
   #:use-module (gnu packages linux)
-  #:use-module (gnu packages specifications))
+  #:use-module (gnu packages specifications)
+  #:use-module (gnu packages xdisorg)
+  #:use-module (gnu packages xorg))
 
 ;;; Commentary:
 ;;;
@@ -958,6 +960,39 @@ (define-public go-github-com-asaskevich-govalidator
 @url{https://github.com/chriso/validator.js,validator.js}.")
     (license license:expat)))
 
+(define-public go-github-com-atotto-clipboard
+  (package
+    (name "go-github-com-atotto-clipboard")
+    (version "0.1.4")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/atotto/clipboard")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0ycd8zkgsq9iil9svhlwvhcqwcd7vik73nf8rnyfnn10gpjx97k5"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/atotto/clipboard"
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'start-xorg-server
+            (lambda* (#:key inputs #:allow-other-keys)
+              ;; The test suite requires a running X server.
+              (system "Xvfb :1 &")
+              (setenv "DISPLAY" ":1"))))))
+    (native-inputs
+     (list xorg-server-for-tests))
+    (propagated-inputs (list xclip))
+    (home-page "https://github.com/atotto/clipboard")
+    (synopsis "Clipboard for Golang")
+    (description
+     "@code{clipboard} provides copying and pasting to the clipboard for Go.")
+    (license license:bsd-3)))
+
 (define-public go-github-com-audriusbutkevicius-recli
   (package
     (name "go-github-com-audriusbutkevicius-recli")
-- 
2.46.0

